### DISCUSSION TITLE
DeferredRender only appears to render the block

### DISCUSSION THREAD

**Discussion by TheRealNeil** [yippee-fun/phlex#584] [Help] [answered]
Hi,
I tried to implement a patten similar to the `DeferredRender` example in the [docs](https://www.rubydoc.info/gems/phlex/Phlex/DeferredRender) I built a `TabsComponent`
```
class TabsComponent < ApplicationComponent
  include Phlex::DeferredRender
  Tab = Data.define(:name, :content)

  def initialize
    @tabs = []
  end

  def template
    @tabs.each { |t| a { t.name } }
    @tabs.each { |t| article(&t.content) }
  end

  def tab(name, &content)
    @tabs << Tab.new(name, content)
  end
end
```
and rendered it in a rails view with
```
<%= render TabsComponent.new do |t| %>
  <% t.tab(name: 'tab1') do %>
    <p>Tab content!</p>
  <% end %>
<% end %>
```
The html that was output contained an empty `<a>` tag.  I was expecting to see the name "tab1" in there.
```
<!-- Before TabsComponent -->
<a></a>
<article>
  <p>Tab content!</p>
</article>
```
I tried with a `<div>` and `<h1>` tag also without success.
Is there a better way to do this?

**Comment by joeldrapper** [comment]
That's really strange. Your code looks good to me.

Have you tried putting `binding.irb` inside the `tabs.each` loop to inspect what `t` looks like at that point?

You could also try rendering an `a` with a string literal to see if that works and then pinpoint where it's going wrong.

**Reply by TheRealNeil** [reply, marked as answer]
`t.name` appears to be a `Hash`
```
0> t.inspect
=> "#<data TabsComponent::Tab name={:name=>\"tab1\"}, content=#<Proc:0x0000000113c2b460 /.../gems/phlex-rails-1.0.0/lib/phlex/rails/unbuffered_overrides.rb:11>>"

0> t.name
=> {:name=>"tab1"}
```
so
```
@tabs.each do |t|
  a { t.name[:name] }
end
```
worked and so did the string literal.

My bad, I had not changed the example `name` parameter to a named parameter but I called it as such 😞 
```
def tab(name:, &content)
  @tabs << Tab.new(name:, content)
end
```
Thanks for the quick reply and prompt to debug.

**Reply by joeldrapper** [reply]
That's weird. I thought `Data` objects allowed for positional or named parameters.
